homework/list.js

```diff
@@ -1,79 +1,92 @@
 'use strict';
 
-(function() {
+(function(exports) {
+	var TodoListManager = function() {
+		this._listNoteContent = [];
+		this._wrapper = document.querySelector('#note-list-wrapper');
+	};
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+	TodoListManager.prototype = {
+		start() {
+			this.fetchList(function(data) {		
+				this.updateList(data)
+				.then((function(){
+					this.drawList(); 
+				}).bind(this)).then((function(){
+					this.preloadFirstNote();
+				}).bind(this)).catch((function(error){
+				}).bind(this));			
+			}.bind(this));
+			
+			
+			window.addEventListener('click', function(event) {
+				this.onNoteOpen(event);
+			}.bind(this));
+		},
+		 
+		onNoteOpen(event) {
```

Comment:
Well it's a convention to prefix with `on` to handle the corresponding event, rather than to fire it. So, you should rename this function as `onClick` and to dispatch the `note-open`.